### ISSUE TITLE
Error while running ESLint: Invalid regular expression flags

### ISSUE THREAD

**Issue by eakl** [eslint/eslint#12538]
**Tell us about your environment**

* **ESLint Version:** 6.6.0
* **Node Version:** 12.13.0
* **npm Version:** 6.12.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```yaml
env:
  browser: true
  es6: true
  node: true
extends:
  - standard
globals:
  Atomics: readonly
  SharedArrayBuffer: readonly
parserOptions:
  ecmaFeatures:
    jsx: true
  ecmaVersion: 2018
  sourceType: module
plugins:
  - react
rules: {}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**
Creating any `*.js` file trigger the error. My linter in Atom report the following

**What did you expect to happen?**
No linter error flag when a JS file is created

**What actually happened? Please include the actual, raw output from ESLint.**
```
Error while running ESLint: Invalid regular expression flags
Invalid regular expression flags
Referenced from: /Users/me/dev/.eslintrc.yml

/Users/me/dev/node_modules/eslint/lib/source-code/source-code.js:426
return /\s/u.test(text.replace(/\/\*.*?\*\//gus, ""));
                               ^
```

**Comment by terrenmoe** [comment]
In the v4.19.1, I experienced the same error from the file and line at 
/lib/source-code/source-code.js:426

The return statement of the function isSpaceBetweenTokens on the return line the following code
 `return /\s/u.test(text.replace(/\/\*.*?\*\//gus, ""));`
should be
 `return /\s/u.test(text.replace(/\/\*.*?\*\//, "gus"));`

This solved the problem for me but the function in the most recent src seems to be depreciated.

**Comment by kostia** [comment]
Same here:
```
/Users/kostia/Code/scrivito_js/js/node_modules/eslint/lib/source-code/source-code.js:426
        return /\s/u.test(text.replace(/\/\*.*?\*\//gus, ""));
                                       ^

SyntaxError: Invalid regular expression flags
  at createScript (vm.js:80:10)
  at Object.runInThisContext (vm.js:139:10)
  at Module._compile (module.js:599:28)
  at Object.Module._extensions..js (module.js:646:10)
  at Module.load (/Users/kostia/Code/scrivito_js/js/node_modules/coffeescript/lib/coffee-script/register.js:45:36)
  at tryModuleLoad (module.js:497:12)
  at Function.Module._load (module.js:489:3)
  at Module.require (module.js:579:17)
  at require (internal/module.js:11:18)
  at Object.<anonymous> (/Users/kostia/Code/scrivito_js/js/node_modules/eslint/lib/source-code/index.js:4:17)
  at Module._compile (module.js:635:30)
  at Object.Module._extensions..js (module.js:646:10)
  at Module.load (/Users/kostia/Code/scrivito_js/js/node_modules/coffeescript/lib/coffee-script/register.js:45:36)
  at tryModuleLoad (module.js:497:12)
  at Function.Module._load (module.js:489:3)
  at Module.require (module.js:579:17)
  at require (internal/module.js:11:18)
  at Object.<anonymous> (/Users/kostia/Code/scrivito_js/js/node_modules/eslint/lib/linter/linter.js:25:22)
  at Module._compile (module.js:635:30)
  at Object.Module._extensions..js (module.js:646:10)
  at Module.load (/Users/kostia/Code/scrivito_js/js/node_modules/coffeescript/lib/coffee-script/register.js:45:36)
  at tryModuleLoad (module.js:497:12)
  at Function.Module._load (module.js:489:3)
  at Module.require (module.js:579:17)
  at require (internal/module.js:11:18)
  at Object.<anonymous> (/Users/kostia/Code/scrivito_js/js/node_modules/eslint/lib/linter/index.js:3:20)
  at Module._compile (module.js:635:30)
  at Object.Module._extensions..js (module.js:646:10)
  at Module.load (/Users/kostia/Code/scrivito_js/js/node_modules/coffeescript/lib/coffee-script/register.js:45:36)
  at tryModuleLoad (module.js:497:12)
  at Function.Module._load (module.js:489:3)
  at Module.require (module.js:579:17)
  at require (internal/module.js:11:18)
  at Object.<anonymous> (/Users/kostia/Code/scrivito_js/js/node_modules/eslint/lib/cli-engine/cli-engine.js:25:20)
  at Module._compile (module.js:635:30)
  at Object.Module._extensions..js (module.js:646:10)
  at Module.load (/Users/kostia/Code/scrivito_js/js/node_modules/coffeescript/lib/coffee-script/register.js:45:36)
  at tryModuleLoad (module.js:497:12)
  at Function.Module._load (module.js:489:3)
  at Module.require (module.js:579:17)
  at require (internal/module.js:11:18)
  at Object.<anonymous> (/Users/kostia/Code/scrivito_js/js/node_modules/eslint/lib/cli-engine/index.js:3:23)
  at Module._compile (module.js:635:30)
  at Object.Module._extensions..js (module.js:646:10)
  at Module.load (/Users/kostia/Code/scrivito_js/js/node_modules/coffeescript/lib/coffee-script/register.js:45:36)
  at tryModuleLoad (module.js:497:12)
  at Function.Module._load (module.js:489:3)
  at Module.require (module.js:579:17)
  at require (internal/module.js:11:18)
  at Object.<anonymous> (/Users/kostia/Code/scrivito_js/js/node_modules/eslint/lib/api.js:8:23)
  at Module._compile (module.js:635:30)
  at Object.Module._extensions..js (module.js:646:10)
  at Module.load (/Users/kostia/Code/scrivito_js/js/node_modules/coffeescript/lib/coffee-script/register.js:45:36)
  at tryModuleLoad (module.js:497:12)
  at Function.Module._load (module.js:489:3)
  at Module.require (module.js:579:17)
  at require (internal/module.js:11:18)
  at ForkTsCheckerWebpackPlugin.validateEslint (/Users/kostia/Code/scrivito_js/js/node_modules/fork-ts-checker-webpack-plugin/lib/index.js:173:23)
  at new ForkTsCheckerWebpackPlugin (/Users/kostia/Code/scrivito_js/js/node_modules/fork-ts-checker-webpack-plugin/lib/index.js:88:53)
  at module.exports (/Users/kostia/Code/scrivito_js/js/webpack.config.js:152:7)
  at module.exports.karma (/Users/kostia/Code/scrivito_js/js/karma/base.js:15:19)
  at module.exports.karma (/Users/kostia/Code/scrivito_js/js/karma.conf.js:10:3)
  at Object.parseConfig (/Users/kostia/Code/scrivito_js/js/node_modules/karma/lib/config.js:395:5)
  at new Server (/Users/kostia/Code/scrivito_js/js/node_modules/karma/lib/server.js:67:24)
  at Object.exports.run (/Users/kostia/Code/scrivito_js/js/node_modules/karma/lib/cli.js:295:7)
  at Object.<anonymous> (/Users/kostia/Code/scrivito_js/js/node_modules/karma/bin/karma:3:23)
  at Module._compile (module.js:635:30)
  at Object.Module._extensions..js (module.js:646:10)
  at Module.load (module.js:554:32)
  at tryModuleLoad (module.js:497:12)
  at Function.Module._load (module.js:489:3)
  at Function.Module.runMain (module.js:676:10)
  at startup (bootstrap_node.js:187:16)
  at bootstrap_node.js:608:3
```

**Comment by phorsuedzie** [comment]
See also: #12493. Please re-check node version used.

**Comment by kostia** [comment]
@eakl Are you sure that you are using Node-12? Because  I only had this issue with an older version of Node (in my particular case with Node-8.9). Installing a recent Node (12) fixed the issue for me.

That being said, if `eslint` relies on certain recent regexp flags, it should restrict the Node version in `package.json` with `engines`.

**Comment by kaicataldo** [comment]
This looks like a duplicate of https://github.com/eslint/eslint/issues/12493.

@eakl Can you confirm you're seeing this with Node `12.13.0`?

> That being said, if eslint relies on certain recent regexp flags, it should restrict the Node version in package.json with engines.

`engines` should be correctly specified in ESLint's `package.json`, but please let us know if you see something wrong.

**Comment by eakl** [comment]
Correct, I'm using Node v12.13.0 through nvm

**Comment by miachenmtl** [comment]
Same issue, using Node v12.8.0. Works fine with `npx eslint *.js`, but there's an error loading eslint in Atom. But downgrading to eslint@6.5.0 makes Atom work fine.

**Comment by kaicataldo** [comment]
@eakl Can you confirm that this behavior also occurs on the command line?

Both you and @miachenmtl seem to be using Atom, so it seems more likely that this is an issue with the Atom editor integration.

**Comment by anatoly314** [comment]
I have the same issue with Node v12.13.0 using Webstorm

**Comment by kaicataldo** [comment]
@anatoly314 Can you confirm whether or not you see this same issue when running ESLint from the command line? If not, this is an issue with the editor integration.

**Comment by anatoly314** [comment]
I confirm that I have same issue from the command line.
I tried to run the following command `eslint "src/Components/Tick.js" `
and get the following error:

```
/Users/anatoly/Documents/sandbox-projects/react-playground/node_modules/eslint/lib/source-code/source-code.js:426
        return /\s/u.test(text.replace(/\/\*.*?\*\//gus, ""));
                                       ^

SyntaxError: Invalid regular expression flags
    at NativeCompileCache._moduleCompile (/Users/anatoly/Documents/sandbox-projects/react-playground/node_modules/v8-compile-cache/v8-compile-cache.js:242:18)
    at Module._compile (/Users/anatoly/Documents/sandbox-projects/react-playground/node_modules/v8-compile-cache/v8-compile-cache.js:186:36)
    at Object.Module._extensions..js (module.js:646:10)
    at Module.load (module.js:554:32)
    at tryModuleLoad (module.js:497:12)
    at Function.Module._load (module.js:489:3)
    at Module.require (module.js:579:17)
    at require (/Users/anatoly/Documents/sandbox-projects/react-playground/node_modules/v8-compile-cache/v8-compile-cache.js:161:20)
    at Object.<anonymous> (/Users/anatoly/Documents/sandbox-projects/react-playground/node_modules/eslint/lib/source-code/index.js:4:17)
    at Module._compile (/Users/anatoly/Documents/sandbox-projects/react-playground/node_modules/v8-compile-cache/v8-compile-cache.js:194:30)
Anatolys-MacBook-Pro-2:react-playground anatoly$ 

```

**Comment by anatoly314** [comment]
I think that I know how to solve the issue, at least with WebStorm it works. It looks like Webstorm grabs on startup default Node/NPM version. If you manage multiple Node/NPM versions by nvm and the default one is `v8.9.1` like in my scenario it won't work. 

To fix it you need to go to: `Webstorm -> Preferences -> Languages and Frameworks -> Node.js and NPM` and select correct `Node interpreters`, see screenshot below:

<img width="1020" alt="Screen Shot 2019-11-17 at 18 47 21" src="https://user-images.githubusercontent.com/1641594/69010709-e1574e00-096a-11ea-9957-97377a131902.png">

**Comment by anatoly314** [comment]
And of course, in the command line, it worked without any issue with correct Node.JS/NPM :)

**Comment by eakl** [comment]
@kaicataldo Indeed, no error message when running ESLint from the command line.

**Comment by kaicataldo** [comment]
Thanks for getting back to us! It does sound like an issue with the VS Code editor integration to me. If you’re unable to solve it, could you make an issue on the VS Code ESLint plugin repo? For what it’s worth, I use VS Code with the ESLint plugin and have not run into this issue.

**Comment by leeDav** [comment]
For me, this turned out to be _something_ to do with `nvm`, updating it to the latest version got it working again.

I can't recall if I deleted `node_modules` and reinstalled as I'd done that several times in debugging attempts.

My project, using create react app, had an extra problem of missing `browserlist` package (peer dep of `postcss-browser-comments` but manually adding that as in `devDeps` worked.

**Comment by ljharb** [comment]
I very much doubt this is related to nvm specifically; perhaps just to running the wrong node version.

**Comment by adelriosantiago** [comment]
Any update on this? Why is this error happening?

**Comment by adelriosantiago** [comment]
> In the v4.19.1, I experienced the same error from the file and line at
> /lib/source-code/source-code.js:426
> 
> The return statement of the function isSpaceBetweenTokens on the return line the following code
> `return /\s/u.test(text.replace(/\/\*.*?\*\//gus, ""));`
> should be
> `return /\s/u.test(text.replace(/\/\*.*?\*\//, "gus"));`
> 
> This solved the problem for me but the function in the most recent src seems to be depreciated.

I don't think this is an appropriate solution, the regex itself is perfectly good. And it is replacing the match with an empty string `""`, the line you suggest would change that to `gus`

**Comment by natcodes** [comment]
I have the same issue when I used create-react-app and opened the app in Atom. The difference is I have an error flag (Error while running ESLint: Invalid regular expression flags) on the first line of every file even if it is empty. The program will run fine but the ugly error is there. 
**What I've tried:** 
I updated to Node 12 and it didn't work. I have another react app I had been working on for a year now that had an older version of ESlint so I decided to downgrade it and see if that would fix it since my old app had no such error. In doing so I got an error telling me I can not manually downgrade when using create-react-app so I had to remove and reinstall the package-lock.json and node modules to make it whole again. I tried using ESlint ignore line but no luck. I don't want the eyesore and I can't seem to be able to fix it. If anyone has a solution, hmu.

**Comment by mdjermanovic** [comment]
Editor/plugin/extension is running ESLint in an older Node version. I guess that their support might be able to help with this issue.

For Atom, I believe it's this repo: https://github.com/AtomLinter/linter-eslint

A similar issue with VSCode was resolved by updating VSCode to the latest version (see #12620).

**Comment by platinumazure** [comment]
At this point, I don't think ESLint is doing anything wrong here. We only support a specific range of Node versions, and some editor integration are running the latest ESLint in a version of Node that we don't support.

As has been mentioned throughout the thread, it is appropriate to open an issue in the issue tracker for your editor integration, if you run into this.

I'm going to close the issue since this isn't an issue with ESLint. ESLint team, please reopen if a bug is identified in ESLint.

**Comment by damien-git** [comment]
The related `lint-eslint` plugin issue is at https://github.com/AtomLinter/linter-eslint/issues/1306 .
This led me to update Atom to version 1.42, which fixed the problem.
